csharp/ql/test/query-tests/Nullness/Implications.expected

```diff
@@ -979,6 +982,7 @@
 | D.cs:221:13:221:14 | access to local variable o4 | null | D.cs:220:18:220:26 | ... + ... | null |
 | D.cs:221:13:221:22 | ... == ... | false | D.cs:221:13:221:14 | access to local variable o4 | non-null |
 | D.cs:221:13:221:22 | ... == ... | true | D.cs:221:13:221:14 | access to local variable o4 | null |
+| D.cs:221:13:221:22 | ... == ... | true | D.cs:221:19:221:22 | null | non-null |
```

Comment:
This doesn't seem right to me.

Reply:
Yeah, looks weird, but [`o4`](https://github.com/Semmle/ql/blob/master/csharp/ql/test/query-tests/Nullness/D.cs#L220) is always non-`null`, because ["In string concatenation operations, the C# compiler treats a null string the same as an empty string."](https://docs.microsoft.com/en-us/dotnet/csharp/how-to/concatenate-multiple-strings). Therefore, if the equality check [`o4 == null`](https://github.com/Semmle/ql/blob/master/csharp/ql/test/query-tests/Nullness/D.cs#L221) is `true` (which it never is), then `null` must be non-`null`.